### ISSUE TITLE
Regular messages originating from endpoints declared on a transport different than the message source cannot be transferred   

### ISSUE THREAD

**Issue by tmasternak** [Particular/NServiceBus.MessagingBridge#522]
### Description

Whenever the bridge transfers a regular message from one transport to another it (among other things) does the `ReplyToAdress` header translation. This is needed to ensure that the physical address stored in that header corresponds to an existing queue on the target transport. 

In some situations (described in "Who's affected" section) this translation fails and messages end up in the bridge error queue.

### Symptoms

An exception similar to the one below is logged by the bridge, and the affected messages are moved to the bridge error queue.

```code
System.Exception: Failed to shovel message for endpoint <endpoitn-name> with id <message-id> from <source-transport> to <target-transport>
 ---> System.Exception: No target address mapping could be found for source address: <physical-on-the-source-transport>. Ensure names have correct casing as mappings are case-sensitive. Nearest configured match: <some-other-physical-address-on-the-source-transport>
   at EndpointRegistry.TranslateToTargetAddress(String sourceAddress) in /_/src/NServiceBus.MessagingBridge/EndpointRegistry.cs:line 67
   at MessageShovel.TransformRegularMessageReplyToAddress(TransferContext transferContext, OutgoingMessage messageToSend, IEndpointRegistry targetEndpointRegistry) in /_/src/NServiceBus.MessagingBridge/MessageShovel.cs:line 105
   at MessageShovel.TransferMessage(TransferContext transferContext, CancellationToken cancellationToken) in /_/src/NServiceBus.MessagingBridge/MessageShovel.cs:line 76
   --- End of inner exception stack trace ---
   at MessageShovel.TransferMessage(TransferContext transferContext, CancellationToken cancellationToken) in /_/src/NServiceBus.MessagingBridge/MessageShovel.cs:line 76
   at NServiceBus.Transport.SqlServer.ProcessStrategy.TryHandleMessage(Message message, TransportTransaction transportTransaction, ContextBag context, CancellationToken cancellationToken) in /_/src/NServiceBus.Transport.SqlServer/Receiving/ProcessStrategy.cs:line 53
   at NServiceBus.Transport.SqlServer.ProcessWithNativeTransaction.TryProcess(Message message, TransportTransaction transportTransaction, ContextBag context, CancellationToken cancellationToken) in /_/src/NServiceBus.Transport.SqlServer/Receiving/ProcessWithNativeTransaction.cs:line 119
```

### Who's affected

NOTE: this is not an exhaustive list of problematic scenarios. Affected users will see a log entry similar to the one defined in the `Symptoms` section. 

The bug affects customers in various scenarios. In particular, when

* a message is retired from ServiceControl after source and destination endpoints have been migrated since the failure happened,
* a message is picked up by the bridge when users are deploying physical instances of the same logical endpoints (source and destination) and the bridge wins the race with existing physical instances on the source side,
* there are some messages in the destination endpoint input queue and before all messages are processed the bridge is configured to treat that queue and the queue stored in the `ReplyToAddress` of the remaining messages as proxy queues
* an endpoint sends a delayed message to some other endpoint and both endpoints are migrated to the other side of the bridge before the message is delivered to the destination queue

### Root cause

The bug is caused by a false assumption that exists in the implementation of the `ReplyToAddress` header translation logic. The logic assumes that the `ReplyToAddress` header value can only point to an endpoint that is declared on the same transport as the one from which the message was picked. There are however numerous scenarios in which this condition does not hold

**Comment by jpalac** [comment]
Will be released in 4.0.0

**Comment by jpalac** [comment]
Release [4.0.1](https://github.com/Particular/NServiceBus.MessagingBridge/releases/tag/4.0.1) is available.
